pkg/sysregistriesv2/shortnames.go

```diff
@@ -0,0 +1,287 @@
+package sysregistriesv2
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/homedir"
+	"github.com/containers/storage/pkg/lockfile"
+	"github.com/pkg/errors"
+)
+
+// defaultShortNameMode is the default mode of registries.conf files if the
+// corresponding field is left empty.
+const defaultShortNameMode = types.ShortNameModePermissive
+
+// userShortNamesFile is the user-specific config file to store aliases.
+var userShortNamesFile = filepath.FromSlash("containers/short-name-aliases.conf")
+
+// shortNameAliasConf is a subset of the `V2RegistriesConf` format.  It's used in the
+// software-maintained `userShortNamesFile`.
+type shortNameAliasConf struct {
+	// A map for aliasing short names to their fully-qualified image
+	// reference counter parts.
+	Aliases map[string]string `toml:"aliases"`
+
+	namedAliases map[string]reference.Named
+}
+
+// ResolveShortNameAlias performs an alias resolution of the specified name.
+// The user-specific short-name-aliases.conf has precedence over aliases in the
+// assembled registries.conf.  It returns whether the provided name could be
+// resolved to an alias, the possibly resolved alias, and an error.
```

Comment:
It could be determined by `nil` check but I find the code much easier to read with:
```golang
resolved, named, err := ...

if resolved { ... }
```

Reply:
OTOH
```go
value, err := …
if value != nil { use value }
```
seems natural — and it makes it trivially visible that `value != nil` implies “`value` is safe to dereference”.

I guess this is not blocking, but `resolved == (named != nil)`, the documentation should clearly say so.

Reply:
You convinced me. I'll change :+1: